### PR TITLE
feat(outputs): Add plugin for nebius cloud monitoring

### PR DESCRIPTION
**Pull request by abrekhov**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go
Comment (srebhan) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go:129]: How about having a local test instance?
```suggestion
			plugin := &NebiusCloudMonitoring{
				EndpointURL:       url,
				MetadataTokenURL:  metadataTokenURL,
				MetadataFolderURL: metadataFolderURL,
				Log:               testutil.Logger{},
			}			
```

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go
Comment (srebhan) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go:135]: You can write this more compact...
```suggestion
			require.NoError(t, plugin.Connect())
			require.NoError(t, plugin.Write(tt.metrics))
```

**Review by srebhan — COMMENTED** [review]
Thanks for your contribution @abrekhov! The code looks quite good already, I have some comments though... One thing I do not get is the authentication handling, it would be nice if you enlighten me. :-)

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:70]: It is classic loopback interface for metdata. (https://nebius.com/il/docs/compute/operations/vm-connect/auth-inside-vm#auth-inside-vm)
```bash
curl -H Metadata-Flavor:Google http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token

{"access_token":"CggVAgAAA...","expires_in":39944,"token_type":"Bearer"}
```

Google Cloud and AWS also use this address for fetching metadata (e.g. https://cloud.google.com/compute/docs/metadata/querying-metadata)

But unfortunately there is no URL yet for metadata. Only IP address.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:81]: I mean that if user set it to zero (or less) - we will preventively switch him to default Request Timeout.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go]: Will this method be invoked necessarily? Do I understand correctly that I also need to update my _test file to invoke this method?

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go]: Sure:
1. First we're fetching our folder id from internal endpoint
2. Than we're obtaining IAMToken (or access token) of service account (that was used to create a VM in the cloud)
3. We're making requests to monitoring endpoint with service account access token and folderID to store telegraf metrics in Nebius Monitoring backend.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go:24]: Please provide an example from other plugin. Do not get it.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:233]: It helps me a lot while I was debugging the final json that was pushed to backend. Are you sure?

**Comment by abrekhov** [comment]
@srebhan Could you please check latest version? I wish to correct everything until 1.27 be released. Thanks!

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:40]: Yes. Delete it and use `time.Now()`directly in the spots where you used this function.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:70]: Hmmm. Should we allow the user to change it? If not, please put a comment next to those declarations explaining why there is a hard-coded IP address (basically what you said above)...

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:80]: Yes I do understand that, but the question is would a timeout of zero make sense to indicate "please wait forever"?

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go]: The method is guaranteed to be invoked by the agent for all plugins that provide it. For the tests you are right, you need to call it manually.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:228]: Yes. Some people run Telegraf in debug mode in production and logging those kind of information is really spamming the logs. IF you really need this, please add a (boolean) option to the plugin e.g. `dump_output_metrics` and hide that output behind this flag.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:107]: Please remove this line as the error is implicitly declared two lines below...
```suggestion
```

**Review by srebhan — COMMENTED** [review]
@abrekhov thanks for the nice update. I think we are close... Two comments and the ones still open from last time...

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:70]: No. It should not be changed. Adding a comment.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:80]: I think that forever timeout is counter productive :) 
I cannot come up with a case where such timeout will be useful.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:119]: True

File: plugins/outputs/nebius_cloud_monitoring/sample.conf
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/sample.conf:10]: If a user is not suppose to change this, then why is this an option at all? 

What is the impact of a user changing this?

I'm asking because I would hate to get a bug that comes in where someone says they changed this and in reality it is the backend rejecting metrics because of the change.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:225]: do we need this printed at every send? I don't mind having the endpoint printed during plugin startup

File: plugins/outputs/nebius_cloud_monitoring/README.md
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/README.md:32]:
```suggestion
This plugin currently only supports Compute metadata based authentication.
```

File: plugins/outputs/nebius_cloud_monitoring/README.md
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/README.md:37]:
```suggestion
Folder ID from instance metadata.
```

I don't like TODOs in code ;)

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:169]: I see this is required for Google, how does AWS react to this? Does it just ignore it?

File: plugins/outputs/nebius_cloud_monitoring/README.md
Comment (powersj) [reply plugins/outputs/nebius_cloud_monitoring/README.md:32]: Do you also want to specify which clouds?

**Review by powersj — COMMENTED** [review]
Thank you for this! Looks nice, few questions inline.

We would love to land this this week, which is why I am not waiting for Sven to mark this as ready for final review to do a review ;)

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go]: I've added some info to README about it. This internal endpoint is only accessible from inside of VM and the information (folder id and token) is specific to owner of VM.

File: plugins/outputs/nebius_cloud_monitoring/sample.conf
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/sample.conf:10]: You absolutely right. We need to delete it. It was created for the future development of Nebius Monitoring Backend. Now it stale a bit. So changing of this option will lead to failure.

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (abrekhov) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:169]: On Nebius Cloud Platform both of methods of getting a metadata implemented (through google like and aws like style). I decided to do it through gcp style.

File: plugins/outputs/nebius_cloud_monitoring/README.md
Comment (powersj) [inline plugins/outputs/nebius_cloud_monitoring/README.md:38]: This should clean up the linter, there is an extra space on the end of 36 and 37 is too long.

```suggestion
Folder ID from instance metadata. In this plugin we use [Google Cloud notation]
This internal metadata endpoint is only accessible for VMs from the cloud.

[Google Cloud notation]: https://nebius.com/il/docs/compute/operations/vm-info/get-info#gce-metadata)
```

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (powersj) [reply plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:170]: Ah thank you for the explination

**Comment by powersj** [comment]
I think removing service config option is all that is left and I can give it the +1

**Review by powersj — APPROVED** [review]
Thank you!

File: plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go
Comment (srebhan) [inline plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go:109]:
```suggestion
	a.Log.Debugf("Getting folder ID in %s", a.metadataFolderURL)
```

**Review by srebhan — APPROVED** [review]
Looks good to me. I have two more comments on logging output but nothing that should hold-up this PR...

**Comment by powersj** [comment]
Thank you all!

**Comment by abrekhov** [comment]
Thank you for review guys!

@srebhan @powersj

**Pull request merged**
